lib/matplotlib/image.py

```diff
@@ -338,6 +339,20 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
             out_width = int(out_width_base)
             out_height = int(out_height_base)
 
+        # decide if we are going to do anti-aliasing or not on the image
+        interpolation = self.get_interpolation()
+        if interpolation == 'antialiased':
+            # default
+            interpolation = 'nearest'
+            # do antialiasing....
+            # compare the number of displayed pixels to the number of
+            # the data pixels.
+            samplerate = t.transform([A.shape[1], 0])[0] / A.shape[1]
+            if (samplerate < 3 and
+                np.abs(int(samplerate) - (samplerate)) > 0.0001):
```

Comment:
Instead of having a floating point slop here: the resampling will transform the image from some integer number of data points (m0, n0) to some integer number of pixels (m1, n1).  I would perform antialiasing if (m1 < 2*m0 (exact, integer comparison) and m1 != m0 (exact, integer equality)) and likewise for n (or perhaps replace 2 by 3 and adjust accordingly).  (The point is just that the real quantities we care about here are integers, so we don't need to involve floating point slop.)